### ISSUE TITLE
Static Web App deployment fails - No matching static site was found or the api key was invalid

### ISSUE THREAD

**Issue by dgard1981** [Azure/static-web-apps#126]
I have created a Static Web App from the portal, but when I create a PR to run the action I see an error -

```
The content server has rejected the request with: BadRequest
Reason: No matching static site was found or the api key was invalid.
```

I have deleted the Static Web App and recreated it, and the issue remains.

The Static Web App was created against the 'master' branch, and I have copied the action to a feature branch that I am working on. From the 'Environments' tab it says "Staging environments are automatically created when a pull request is generated", so I was expecting a Stating environment to be created.

Here is the GitHub action that was created. I have slightly altered the `on.push` and `on.pull_request` properties to reflect my desired triggers.

In my case the static website is truly static, so the optional `api_location` and `app_artifact_location` properties have been commented out.

Please note that commits to 'master' do result in a successful deployment to the production environment - it only seems to be the staging environments that have issues.

```
name: Azure Static Web Apps CI/CD

on:
  push:
    branches:
      - master
    paths:
      - 'assets/**'
  pull_request:
    types: [opened, synchronize, reopened, closed]
    branches:
      - '**'
      - '!master'
    paths:
      - 'assets/**'

jobs:
  build_and_deploy_job:
    if: github.event_name == 'push' || (github.event_name == 'pull_request' && github.event.action != 'closed')
    runs-on: ubuntu-latest
    name: Build and Deploy Job
    steps:
      - uses: actions/checkout@v2
        with:
          submodules: true
      - name: Build And Deploy
        id: builddeploy
        uses: Azure/static-web-apps-deploy@v0.0.1-preview
        with:
          azure_static_web_apps_api_token: ${{ secrets.AZURE_STATIC_WEB_APPS_API_TOKEN_WHITE_SMOKE_061101503 }}
          repo_token: ${{ secrets.GITHUB_TOKEN }} # Used for Github integrations (i.e. PR comments)
          action: "upload"
          ###### Repository/Build Configurations - These values can be configured to match you app requirements. ######
          # For more information regarding Static Web App workflow configurations, please visit: https://aka.ms/swaworkflowconfig
          app_location: "/assets" # App source code path
          ### api_location: "api" # Api source code path - optional
          ### app_artifact_location: "" # Built app content directory - optional
          ###### End of Repository/Build Configurations ######

  close_pull_request_job:
    if: github.event_name == 'pull_request' && github.event.action == 'closed'
    runs-on: ubuntu-latest
    name: Close Pull Request Job
    steps:
      - name: Close Pull Request
        id: closepullrequest
        uses: Azure/static-web-apps-deploy@v0.0.1-preview
        with:
          azure_static_web_apps_api_token: ${{ secrets.AZURE_STATIC_WEB_APPS_API_TOKEN_WHITE_SMOKE_061101503 }}
          action: "close"
```

**Comment by miwebst** [comment]
So the problem here could be related to the pull request base branch (where to merge to) not being the 'master' branch that the static web app was configured against. The error you are seeing is indicative of our service not recognizing the branch associated with this deployment.

Here are some simple steps that create a staging environment:

1. Create static web app for some repo
2. Use Github editor to change some file
3. Before commiting file change, select the option to create a new branch and create pull request
4. Create the pull request (verify the base branch is the branch that the static web app is connected to)
5. Verify that the Github Action is running and creates a preview environment

**Comment by dgard1981** [comment]
> So the problem here could be related to the pull request base branch (where to merge to) not being the 'master' branch that the static web app was configured against.

@miwebst - that is exactly what is happening, and I can confirm that a PR to our 'master' branch creates a staging environment.

So does that mean that the Static Web App will only create a staging environment for PR's or commits to the branch that the Static Web App is configured against? If so, is that expected to chance before the service comes out of preview?

In our case, we have a stable 'master' branch (which the Static Web App is configured against), a 'develop' branch, and then feature/bugfix/hotfix branches. We branch from 'develop' when creating feature/bugfix/hotfix branches, and then merge back in to 'develop', which in turn is merged into 'master'. So the way that Static Web Apps currently work means that we'll see lots of GitHub action failures, and that we won't actually be able to test our changes in the Static Web App.

Thanks,
David

**Comment by miwebst** [comment]
I see, that makes sense. Today it will only create the PR environment if the PR is going against the branch the Static Web App was created against. We have something coming that will help make your scenario a valid one (all you will need is the token and it will know you want a staging environment), I dont have an ETA for that work to share right now though.

**Comment by dgard1981** [comment]
@miwebst - Great to know something is on the horizon. At the moment though, as Static Web Apps only work if we interact with the 'master' branch, unfortunately it's not going to be something we can use in this case :(

**Comment by miwebst** [comment]
Sorry to hear that! To clarify the PR feature only works when creating a pull request to merge into the branch the Static Web App is created against, not just 'master'.

Another option for you may be to create a SWA for your develop branch and one for the master branch. In that case you can still use stage sites when developing in your feature branches and also use them to verify the changes in develop before moving them to master. Its a little more overhead but can unblock you for now.

**Comment by dgard1981** [comment]
> Sorry to hear that! To clarify the PR feature only works when creating a pull request to merge into the branch the Static Web App is created against, not just 'master'.

Yep, understood, I just said 'master' as that is the branch we are using.

Thanks for the replies - I'll keep an eye on the preview to see how the service evolves.

**Comment by mustafa90-D** [comment]
Hi team,

I am trying to deploy a simple static HTML dashboard to Azure Static Web Apps using GitHub Actions.

My workflow YAML looks like this:

name: Azure Static Web Apps CI/CD

on:
  push:
    branches:
      - main
  pull_request:
    types: [opened, synchronize, reopened, closed]
    branches:
      - main

jobs:
  build_and_deploy_job:
    if: github.event_name == 'push' || (github.event_name == 'pull_request' && github.event.action != 'closed')
    runs-on: ubuntu-latest
    name: Build and Deploy Job
    steps:
      - uses: actions/checkout@v3
        with:
          submodules: true

      - name: Build And Deploy
        id: builddeploy
        uses: Azure/static-web-apps-deploy@v1
        with:
          azure_static_web_apps_api_token: ${{ secrets.AZURE_STATIC_WEB_APPS_API_TOKEN_NICE_ISLAND_03C8E470F }}
          action: "upload"
          app_location: "/"   # repo root, where index.html is
          api_location: ""
          output_location: "" # no build, just static files
          skip_app_build: true

  close_pull_request_job:
    if: github.event_name == 'pull_request' && github.event.action == 'closed'
    runs-on: ubuntu-latest
    name: Close Pull Request Job
    steps:
      - name: Close Pull Request
        id: closepullrequest
        uses: Azure/static-web-apps-deploy@v1
        with:
          azure_static_web_apps_api_token: ${{ secrets.AZURE_STATIC_WEB_APPS_API_TOKEN_NICE_ISLAND_03C8E470F }}
          action: "close"


I have correctly set the secret in my repository settings as AZURE_STATIC_WEB_APPS_API_TOKEN_NICE_ISLAND_03C8E470F.

However, the workflow still sometimes fails with:



No matching Static Web App was found or the api key was invalid.
